pkgs/tools/typesetting/tex/mftrace/default.nix

```diff
@@ -0,0 +1,61 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+, makeWrapper
+, autoreconfHook
+, buildEnv
+, python3
+, fontforge
+, potrace
+, texlive
+}:
+
+# to use with a texlive distribution, ensure that the desired fonts and
+# the packages kpathsea, t1utils, metafont are available at runtime
+#
+# mftrace can also be added to nixpkgs' texlive via texlive.combine
+#
+# accepted overrides:
+# - potrace = autotrace
+# - fontforge = ghostscript (limited functionality)
+# - fontforge = null (limited functionality)
+
+let self = stdenv.mkDerivation rec {
+  pname = "mftrace";
+  version = "1.2.20";
+
+  # https://lilypond.org/download/sources/mftrace/mftrace-1.2.20.tar.gz
+  # is incomplete, fetch repo and use autoconf instead
+  # see https://github.com/hanwen/mftrace/issues/13
+  src = fetchFromGitHub {
+    owner = "hanwen";
+    repo = "mftrace";
+    rev = "release/${version}";
+    sha256 = "02ik25aczkbi10jrjlnxby3fmixxrwm2k5r4fkfif3bjfym7nqbc";
+  };
+
+  nativeBuildInputs = [ makeWrapper autoreconfHook python3 potrace ];
+
+  buildInputs = [ fontforge potrace ];
+
+  postInstall = ''
+    wrapProgram $out/bin/mftrace --prefix PATH : ${lib.makeBinPath buildInputs}
+  '';
+
+  passthru.tlType = "bin";
+  passthru.pkgs = [ self ] ++ (builtins.concatMap (builtins.getAttr "pkgs")
+    (with texlive; [ kpathsea t1utils metafont ]));
```

Comment:
I am not sure about this as I don't see such pattern very often.

Reply:
It is supposed to pull all the transitive dependencies of those three packages (based on how it's done in `flatDeps` inside `texlive/default.nix`). The only other packages with `tlType`/`pkgs` are `auctex` and `noweb`, and they do not specify dependencies, so it is indeed a new thing. @veprbl would you be able to confirm that I got this right?

Reply:
Yes, this is how texlive "packages" work. I would write `kpathsea.pkgs ++ t1utils.pkgs ++ metafont.pkgs`, but it's just my preference.